### PR TITLE
Updating Rights Structure

### PR DESCRIPTION
**Pull request by cgalarza**
- Fixing rights form so that the data structure returned is similar to the dynamic field structure.
- Updating form to be smart enough to open and close panels based on the data previous saved
- In order to generate parts of the form I am using canned data for fields that will eventually be retrieved from a graphql endpoint 
- Noted an issue I was having returning the saved object here: HYACINTH-523
- Note: I copied part of the code that we use to generate the metadata form and adapted it slightly  to work with an `onChange` method that receives the previous state. Eventually we will be able to reuse this same classes between the rights form and the metadata form.
- Note: We should deploy this to dev and test manually.

### CODE REVIEW

**Comment by elohanlon** [comment]
Note to @cul/diag reviewers: Feel free to review, but we should hold off on merging this until we resolve HYACINTH-523.

**Comment by cgalarza** [comment]
I think getting this PR merged in might be urgent, and we can fix the bug subsequently.

**Comment by elohanlon** [comment]
@cgalarza Let's see if it's a quick bug fix, but if it ends up being complicated and merging is urgent then I agree.

File: app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/AssetRightsForm.js [hunk 1]
```diff
@@ -1,81 +1,73 @@
...
 
-  return (
-    <>
-      <ContextualNavbar
-        title="Editing Asset Rights"
-        rightHandLinks={[
-          { link: '/digital_objects', label: 'Cancel' },
-        ]}
-      />
+  console.log(rights)
```
Comment (barmintor) [inline app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/AssetRightsForm.js]:
```suggestion
```

File: app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/ItemRightsForm.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
 import { useHistory } from 'react-router-dom';
-import { merge } from 'lodash';
+import { merge, omit, isEmpty, cloneDeep } from 'lodash';
+import { useMutation } from '@apollo/react-hooks';
 
-import keyTransformer from '../../../../utils/keyTransformer';
 import Label from '../../../shared/forms/Label';
 import InputGroup from '../../../shared/forms/InputGroup';
 import BooleanRadioButtons from '../../../shared/forms/inputs/BooleanRadioButtons';
-import { digitalObject as digitalObjectApi } from '../../../../utils/hyacinthApi';
```
Comment (barmintor) [inline app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/ItemRightsForm.js]: 🎉

File: app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/fields/Field.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    return (
+      <InputGroup>
+        <Label align="right">{displayLabel}</Label>
+        {field}
+      </InputGroup>
+    );
+  }
+}
+
```
Comment (barmintor) [inline app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/fields/Field.js]: cool, this component might be useful elsewhere, too?

**Comment by elohanlon** [comment]
@cgalarza Thanks for the digital_object issue investigation / stringify update.  Looks good!

File: app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/ItemRightsForm.js (hunk 2 above)
Comment (elohanlon) [reply app/javascript/hyacinth_v1/components/digital_objects/rights/rights_form/ItemRightsForm.js]: 🎉

**Pull request merged**